src/hist.jl

```diff
@@ -14,6 +16,31 @@ function _check_closed_arg(closed::Symbol, funcsym)
 end
 
 
+## Fast map for tuples (a lot faster than using Base.map)
+@generated function _tuple_map{N}(f, tuples::NTuple{N, Any}...)
```

Comment:
I don't this is the right place for this function. It should be a PR against Base and meanwhile, I think we should use the standard version and live with the slowdown. Is `broadcast` equally slow?